cli/cli
# Enhance gh issue view to conditionally handle v1 projects being available

Closes github/cli#490

This has been quite the adventure and has resulted in a decently sized refactor. I learned a ton along the way and am feeling a lot more confident in my ability to work with the rest of the codebase 🙌 Additionally, while doing this I discovered that v2 project support was never added to this command. I've created a follow-up issue, github/cli#544 to track the remaining work.

## Changes

- Extracted all the printing functionality out of `issue/view.go` and into various issue printer interfaces. This allows for better testing, easier to understand code, and the ability to better tease out the requirements of github/cli#490
- Tests were added/expanded to ensure
  - `gh issue view` will only show v1 projects if available
  - `gh issue view --json projectCards` will only show v1 projects if available, showing no data if not available

## Out of scope

This doesn't enable Project v2 support in `gh issue view`. That work is tracked in github/cli#544

## Testing

- Pull down the branch and run the tests: `go test ./...`
- Build the tool locally using `make`
- Run `bin/gh issue view -R github/cli 490`
- Notice that "Projects" doesn't appear in the output (would be under "assignees")

![Screenshot 2024-09-11 at 10 34 31 AM](https://github.com/user-attachments/assets/eb6dbd08-a19a-4a54-bedd-5ec3e3f4df09)

- Run `bin/gh issue view -R github/cli 490 | cat`
- Notice that there are not projects listed

![Screenshot 2024-09-11 at 10 36 04 AM](https://github.com/user-attachments/assets/379aac29-75de-458e-b53f-a6d5b68a3c48)

"Do the hard thing so you can do the easy fix"
